pkg/controller/controller.go

```diff
@@ -109,6 +108,11 @@ func (haproxy *HAProxyController) UpdateIngressStatus(*extensions.Ingress) []api
 func (haproxy *HAProxyController) ConfigureFlags(flags *pflag.FlagSet) {
 	haproxy.reloadStrategy = flags.String("reload-strategy", "native",
 		`Name of the reload strategy. Options are: native (default), reusesocket or multibinder`)
+	haproxy.maxOldConfigFiles = flags.Int("max-old-config-files", 0,
+		`Maximum old haproxy timestamped config files to allow before being cleaned up. A value <= 0 indicates a single non-timestamped config file will be used`)
+	if *haproxy.maxOldConfigFiles < 0 {
+		*haproxy.maxOldConfigFiles = 0
+	}
```

Comment:
This is just the initialization, the args wasn't parsed yet, so this if should be at OverrideFlags which happens after the parse.

Anyway afaics you are using >0 or <=0 everywhere so it's pointless to check it's value. I think it's safe to just remove it.